app/controllers/api/v1/have_read_books_controller.rb

```diff
@@ -0,0 +1,50 @@
+class Api::V1::HaveReadBooksController < ApplicationController
+  def create
+    have_read_book = HaveReadBook.new(ajax_params)
+    have_read_book.user = current_user
+    have_read_book.build_date(date_params)
+
+    if have_read_book.save
+      to_read_book = ToReadBook.find_by(user: current_user, book_id: ajax_params[:book_id])
+      current_user.update_ranks(ajax_params[:book_id], to_read_book.rank, nil)
```

Comment:
I like this!

I would update the method to have keyword args, i.e.:
current_user.update_ranks(book_id: ajax_params[:book_id], old_rank: to_read_book.rank, new_rank: nil)

And then in the user class
class User
  def update_ranks(book_id:, old_rank:, new_rank:)
    #stuff here
  end
e